test/functional/forknotify.py

```diff
@@ -41,11 +42,17 @@ def run_test(self):
         self.nodes[1].generate(1)
         self.sync_all()
 
-        with open(self.alert_filename, 'r', encoding='utf8') as f:
-            alert_text = f.read()
+        # Give bitcoind 10 seconds to write the alert notification
+        timeout = 10
+        while True:
```

Comment:
yeah; mostly this was motivated by being able to get rid of the stateful `timeout = 10; assert timeout > 0; timeout -= 0.1` combo.
This is also fine.
```python
        for t in xrange(100):
            with open(self.alert_filename, 'r', encoding='utf8') as f:
                alert_text = f.read()
            if alert_text:
                break
            time.sleep(0.1) 
        else:
            raise AssertionError("-alertnotify did not warn of up-version blocks")
```